### PR TITLE
#58772 make the process link endpoints generic

### PR DESCRIPTION
**Pull request by Klaas-Ritense**
Breaking changes:
- PluginProcessLinkResource (moved and renamed to ProcessLinkResource)
- ProcessLinkResource (changed bean name)
- ProcessLinkService (changed bean name)
- PluginProcessLink.id (type id now UUID)
- PluginProcessLink.activityType (type change)
- PluginProcessLinkCreateDto.activityType (type change)
- PluginProcessLinkResultDto.activityType (new property)
- PluginActionDefinition.activityType (type change)
- PluginProcessLinkRepository (No longer extends JpaRepository)
- CopyPluginActionsOnProcessDeploymentListener (Moved & renamed to CopyPluginLinkOnProcessDeploymentListener)

### CODE REVIEW

File: process-link/src/main/kotlin/com/ritense/processlink/security/config/ProcessLinkHttpSecurityConfigurer.kt [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+import org.springframework.http.HttpMethod.POST
+import org.springframework.http.HttpMethod.PUT
+import org.springframework.security.config.annotation.web.builders.HttpSecurity
+
+class ProcessLinkHttpSecurityConfigurer: HttpSecurityConfigurer {
+
+    override fun configure(http: HttpSecurity) {
+        try {
+            http.authorizeRequests()
+                .antMatchers(GET, "/api/v1/process-link").hasAuthority(USER)
```
Comment (theo-ritense) [inline process-link/src/main/kotlin/com/ritense/processlink/security/config/ProcessLinkHttpSecurityConfigurer.kt:33]: Can you explain why this has been changed to 'USER' access?

Comment (Klaas-Ritense) [reply process-link/src/main/kotlin/com/ritense/processlink/security/config/ProcessLinkHttpSecurityConfigurer.kt]: changed it. role_user shouldln't have acces to get/modify process-links

File: process-link/src/main/kotlin/com/ritense/processlink/web/rest/dto/ProcessLinkCreateRequestDto.kt [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+ * limitations under the License.
+ */
+
+package com.ritense.processlink.web.rest.dto
+
+import com.fasterxml.jackson.annotation.JsonTypeInfo
+import com.ritense.processlink.domain.ActivityTypeWithEventName
+
+@JsonTypeInfo(use = JsonTypeInfo.Id.DEDUCTION)
+abstract class ProcessLinkCreateRequestDto (
```
[the remark this answers is not in the document]
Comment (Klaas-Ritense) [reply process-link/src/main/kotlin/com/ritense/processlink/web/rest/dto/ProcessLinkCreateRequestDto.kt]: Changed

**Review by theo-ritense — APPROVED** [review]
I made a small adjustment to remove the processLinkType from the DTO constructors:
https://github.com/valtimo-platform/valtimo-backend-libraries/pull/654/commits/9158f50e12b7c9a7d175d6bf26860926525d4110

**Pull request merged**